apollo-api/src/main/java/com/spotify/apollo/Response.java

```diff
@@ -40,10 +42,33 @@
 
   /**
    * The response headers.
+   * Deprecated in favor of {@link Response#headerEntries()}
    */
+  @Deprecated
   Map<String, String> headers();
 
   /**
+   * The response headers
+   */
+  Iterable<Map.Entry<String, String>> headerEntries();
```

Comment:
I was trying to be conservative and restrictive, conveying the message that this is only for iterating over all the headers. I want to prevent users from looking up keys from whatever object this method returns. Also, I didn't want to lock a specific implementation (we could change the internal implementation for a Map, etc).  But I see now that I chose the wrong interface and it should have been Iterator, but maybe it's too annoying, specially since it does not have size(). What do you think would be better: Iterator or List?